src/js/src/components/SearchBox.vue

```diff
@@ -151,26 +150,29 @@ export default {
       updateSolrSettingGrouping:'updateSolrSettingGrouping',
       updateSolrSettingImgSearch:'updateSolrSettingImgSearch',
       updateSolrSettingUrlSearch:'updateSolrSettingUrlSearch',
-      updateSolrSettingOffset:'updateSolrSettingOffset'
+      updateSolrSettingOffset:'updateSolrSettingOffset',
+      updateFutureSolrSettingGrouping:'updateFutureSolrSettingGrouping',
+      updateFutureSolrSettingImgSearch:'updateFutureSolrSettingImgSearch',
+      updateFutureSolrSettingUrlSearch:'updateFutureSolrSettingUrlSearch',
     }),
     ...mapActions('Notifier', {
       setNotification: 'setNotification'
      
     }),
     handleSubmit() {
       if (this.futureQuery !== this.query ||
-          this.futureGrouped !== this.solrSettings.grouping ||
-          this.futureUrlSearch !== this.solrSettings.urlSearch ||
-          this.futureImgSearch !== this.solrSettings.imgSearch) 
+          this.futureSolrSettings.grouping !== this.solrSettings.grouping ||
+          this.futureSolrSettings.urlSearch !== this.solrSettings.urlSearch ||
+          this.futureSolrSettings.imgSearch !== this.solrSettings.imgSearch) 
```

Comment:
Very long (but not complex) _if_ condition - consider moving it to a separate method so it would be something like if (searchBoxParamsHaveChanged())  { ...

Reply:
My thought exactly. Is being done in SOBACK-30.